llvm/lib/Transforms/Instrumentation/InstrProfiling.cpp

```diff
@@ -148,6 +150,16 @@ cl::opt<bool> SkipRetExitBlock(
     "skip-ret-exit-block", cl::init(true),
     cl::desc("Suppress counter promotion if exit blocks contain ret."));
 
+static cl::opt<bool>
+    SampledInstrument("sampled-instr", cl::ZeroOrMore, cl::init(false),
+                      cl::desc("Do PGO instrumentation sampling"));
+
+static cl::opt<unsigned> SampledInstrumentDuration(
+    "sampled-instr-duration",
```

Comment:
nit: this is more like a sample rate rather than duration?

Reply:
or sampling period.

Reply:
I don't this is the sampling rate. This is burst sampling. In this burst period, the sample rate is 1. this parameter specifies the during for the burst period.

Reply:
If you end up with bursty sampling, maybe `sampled-instr-burst-duration` to be clear. There also needs to be a explanation in comment somewhere to call out burst mode is chosen and explain why.

It can be useful to have a separate `sampled-instr-period`, which defaults to 64K, so that part is also tunable when needed. With that flag, people can set `sampled-instr-burst-duration` to 1, and tune `sampled-instr-period` instead to achieve non-burst sampling. We may give it a try.

Reply:
I like the idea of two flags and allowing users to choose the setting enabling to both bursty or conventional sampling. One suggestion would be to make `sampled-instr-period` a prime number (standard practice in hardware sampling).

Reply:
The current sampling period is implicit to take advantage of  the wrapping behavior to reduce runtime overhead (and code size) -- we don't want to lose that. However I think the implementation can check if the bursty duration is not one, assert that sampling period is not set. Otherwise, generate the non bursty style sampling.